dex-it/src/test/scala/com/wavesplatform/it/sync/MarketOrderTestSuite.scala

```diff
@@ -1,118 +1,427 @@
   }
 
   override protected def beforeAll(): Unit = {
     super.beforeAll()
-    Seq(IssueUsdTx, IssueEthTx) foreach { tx =>
-      node.waitForTransaction(node.broadcastRequest(tx.json.value).id)
+    val txIds = Seq(IssueWctTx, IssueUsdTx, IssueEthTx, IssueBtcTx).map(_.json()).map(node.broadcastRequest(_).id)
+    txIds.foreach(node.waitForTransaction(_))
+    node.waitForTransaction(node.broadcastTransfer(alice, bob.toAddress.toString, 10000.usd, fixedFee, Some(UsdId.toString), None).id)
+  }
+
+  override protected def afterEach(): Unit = {
+    super.afterEach()
+    node.cancelAllOrders(alice)
+    node.cancelAllOrders(bob)
+  }
+
+  def placeOrders(sender: KeyPair, pair: AssetPair, orderType: OrderType, feeMode: String = FIXED.toString)(orders: (Long, Long)*): Unit = {
+    orders.zipWithIndex.foreach {
+      case ((amount, price), idx) =>
+        node.placeOrder(sender = sender,
+                        pair = pair,
+                        orderType = orderType,
+                        amount = amount,
+                        price = price,
+                        version = 3: Byte,
+                        fee = getFee(feeMode),
+                        timestamp = System.currentTimeMillis + idx)
     }
   }
 
-  "Sunny day tests for market orders" in {
+  def placeMarketOrder(sender: KeyPair, pair: AssetPair, orderType: OrderType, amount: Long, price: Long): MatcherStatusResponse = {
+    node.waitOrderStatus(pair, node.placeMarketOrder(node.prepareOrder(sender, pair, orderType, amount, price, fee = fixedFee)).message.id, "Filled")
+  }
+
+  def createAccountWithBalance(balances: (Long, Option[String])*): KeyPair = {
+    val account = KeyPair(ByteStr(s"account-test-${System.currentTimeMillis}".getBytes(StandardCharsets.UTF_8)))
 
-    def placeCounterOrders(sender: KeyPair, pair: AssetPair, ordersType: OrderType)(orders: (Long, Long)*): Unit = {
-      orders.foreach {
-        case (amount, price) =>
-          node.placeOrder(sender, pair, ordersType, amount, price, 0.003.waves).message.id |> (lo => node.waitOrderStatus(pair, lo, "Accepted"))
+    println(s"Created new account. Address: ${account.toAddress}")
+
+    balances.foreach {
+      case (balance, asset) => {
+        if (asset != None)
+          assert(
+            node.assetBalance(alice.toAddress.toString, asset.get.toString).balance >= balance,
+            s"Alice doesn't have enough balance in ${asset.get.toString} to make a transfer"
+          )
+        node.waitForTransaction(node.broadcastTransfer(alice, account.toAddress.toString, balance, fixedFee, asset, None).id)
       }
     }
+    account
+  }
+
+  def getFee(mode: String): Long = {
+    mode match {
+      case "percent" => percentFee.waves
+      case "fixed"   => fixedFee
+      case _         => 0L
+    }
+  }
+
+  def calculateFeeValue(amount: Long, feeMode: String): Long = {
+    feeMode match {
+      case "percent" => amount / 100 * percentFee
+      case "fixed"   => fixedFee
+    }
```

Comment:
Consider using of enum values instead of strings and pattern matching:
```
mode match {
      case DYNAMIC => ...
      case FIXED   => ...
      case PERCENT => ...
    }
```

Reply:
fixed